src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/haflows/HaFlowHistorySpec.groovy

```diff
@@ -0,0 +1,125 @@
+package org.openkilda.functionaltests.spec.flows.haflows
+
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.extension.failfast.Tidy
+import org.openkilda.functionaltests.helpers.HaFlowHelper
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.northbound.dto.v2.haflows.HaFlow
+
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import spock.lang.Ignore
+import spock.lang.Narrative
+import spock.lang.Shared
+
+@Narrative("""Verify that history records are created for the create/update actions.
+History record is created in case the create/update action is completed successfully.""")
+@Slf4j
+class HaFlowHistorySpec extends HealthCheckSpecification {
+
+    @Autowired
+    @Shared
+    HaFlowHelper haFlowHelper
+
+    @Tidy
+    def "History records are created for the create"() {
+        given: "Ha flow"
+        def swT = topologyHelper.getAllNotNeighbouringSwitchTriplets().shuffled().first()
+        def haFlow = haFlowHelper.addHaFlow(haFlowHelper.randomHaFlow(swT))
+
+        when: "Ha Flow history contains create event"
+        //implement method
+        haFlowHelper.getHistory(haFlow.haFlowId).getCreateEntries().size() == 1
+
+        then: "Ha flow possible to be validated"
```

Comment:
Please, be consequent in terms. There should be the one single way to name Ha-Flow. I'd recommend to follow this convention, not 'Ha Flow', 'Ha flow' or anything else.

Reply:
alligned

Reply:
```
def "History records can be get with timestamp filters"() {
        given: "HA-FLow"
        ...
        when: "Delete HA flow"
```
![Oh really?](https://gifdb.com/images/high/oh-really-ace-ventura-scheming-lqxqssaiu0g4n22m.gif)

Reply:
fixed